docs/contribute/DEV_DOCS.md

```diff
@@ -25,6 +25,39 @@ This project strictly adheres to the
 
 Please see the [project board](https://github.com/orgs/nextstrain/projects/6) for currently available issues.
 
+## Installing the development version of Augur
+
+[Install Miniconda with Python 3](https://docs.conda.io/en/latest/miniconda.html).
+If you already have Miniconda installed with Python 2, download the latest Python 3 version and [follow conda's installation instructions](https://conda.io/projects/conda/en/latest/user-guide/install/index.html). Any prior versions of conda should be updated before continuing installation.
+
+Create a new conda environment.
+```bash
+conda create -n nextstrain
```

Comment:
We maintain a `environment.yml` file in the conda repo for dev installs (at least, that's what I use it for!). This will install the dependencies (`mafft` etc) thus avoiding needing to install augur from bioconda and then overwrite it with an editable pip install.

```sh
conda env create -f environment.yml # will call the env `augur`, use `--name nextstrain` to override
conda activate augur
python3 -m pip install -e '.[dev]'
```

If we want to manually create an environment, I believe we'd need `conda create -n nextstrain python=3.8` else we will use the (conda) base python version and thus `python3 -m pip install -e '.[dev]'` will make `augur` available to all conda environments. (Py 36 or 37 would also work.)

Reply:
You're right, @jameshadfield, that the commands as written don't work (my own fault)! @tsibley and I have been working to [improve the Bioconda package](https://github.com/bioconda/bioconda-recipes/pull/26818) and eventually add a Nextstrain CLI package to Bioconda so we can fully phase out the conda YAML. The Bioconda package is more discoverable (you can `conda search -c bioconda augur`) and will also more accurately reflect the pinned versions of dependencies. Ideally, we'd like to have one source of truth for all dependencies; now we have the PyPI package, Bioconda package, and the conda YAML.

After testing this implementation of the docs, a clearer (and more functional) set of steps would be:

 1. Install `mamba` into the `base` environment (`conda install -c conda-forge mamba`)
 2. Create the nextstrain environment with `augur` as an explicit requirement for the environment's dependency resolution. This will create the new environment _and_ install augur with the most appropriate version of Python. (`mamba create -n nextstrain -c conda-forge -c bioconda augur`)
 3. Clone the GitHub repo and pip install the editable version.

These commands work for me, but I'm curious if they work well for you, too, @jameshadfield.